sql/item.cc

```diff
@@ -7035,7 +7035,7 @@ bool Item_default_value::fix_fields(THD *thd, Item **items)
   }
   if (!(def_field= (Field*) sql_alloc(field_arg->field->size_of())))
     goto error;
-  memcpy(def_field, field_arg->field, field_arg->field->size_of());
+  memcpy((void*)def_field, (void*)field_arg->field, field_arg->field->size_of());
```

Comment:
What was the warning fixed here?

Reply:
memcpy is used to copy a class with virtual functions, and clang generates the warning that in this case, it'll copy the vtblptr. Which is fine, but this is the way the warning can be silenced.

Reply:
Isn't memcpy on such object is undefined behavior, i.e. not fine? What do higher versions do?

Reply:
It is, but I think in practice it works in every implementation. 

They refactored this part, by introducing a pure virtual clone method in the Field class, and implementing it in every derived class.

Reply:
OK, that's too much to backport, let's just add the typecasts. Please use C++ style for them